Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -1022,12 +1006,12 @@ private bool ExecuteOnAllServices(Action<IMixedRealityService> execute)
             // If the Mixed Reality Toolkit is not configured, stop.
             if (!HasProfileAndIsInitialized) { return false; }
 
-            foreach (var system in activeSystems)
+            foreach (var system in activeSystems.ToList())
```

Comment:
Is this now going to allocate multiple times a frame on each `Update`, `LateUpdate`, etc?

Reply:
unfortunately, yes. i have not found a safer way to handle this scenario. ideas?

Reply:
If so, I wonder if we should try to follow the pattern in `BaseEventSystem` instead, where it saves off actions (in that case, registering a listener) for a predetermined, known good time in the app cycle

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/0734ab93655113a974682137477cd095d63f87e2/Assets/MixedRealityToolkit/Services/BaseEventSystem.cs#L208

Any profile changes (which I think is what this is aiming to support) could be saved off and handled maybe at the end of update / lateupdate / some known good time, probably near the end of the frame

Reply:
This would be preferable using a static list that is populated/cleared down.  ToList has too many performance overheads